compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,22 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && symname.as_str().eq("Sized")
+                    && this.prev_token.is_lifetime()
+                {
+                    // parse pattern with "'a Sized" we're supposed to give suggestion like
+                    // "'a + Sized"
+                    let mut err = this.struct_span_err(this.prev_token.span, "expected `+` between lifetime and trait");
+                    err.multipart_suggestion(
+                    "missing `+`",
+                     vec![(this.prev_token.span.shrink_to_hi(), " +".to_string()),
+                       (this.token.span.shrink_to_lo(), "".to_string())],
+                     Applicability::MaybeIncorrect,
+                    );
+                    err.emit();
+                    return Err(err);
```

Comment:
You don't need to emit the error if it's being returned right after.

Reply:
I have to return `the err` early. Because at this point, we've parsed 'lifetime + trait which is already incorrect. If we don't return it. The parser would parse more in the following line:
```
2403                 let (pat, colon) = this.parse_fn_param_pat_colon()?;
2404                 if !colon {
```
And it would emit something like:
```
help: if this is a parameter name, give it a type
  |
1 | fn foo<'a>(_: impl 'a Sized: TypeName) {}
  |                            ++++++++++
help: if this is a type, explicitly ignore the parameter name

```
which I think it's not necessary

Reply:
Don't call `emit()` is what I meant, I agree you need to return the `Err` here.

Reply:
this was fixed